src/restricted-endpoints/replication/bulk-document/bulk-document.service.ts

```diff
@@ -96,16 +105,79 @@ export class BulkDocumentService {
     );
     return {
       new_edits: request.new_edits,
-      docs: request.docs.filter((doc) =>
-        this.hasPermissionsForDoc(
-          doc,
-          response.rows.find((responseDoc) => responseDoc.id === doc._id),
-          ability,
+      docs: request.docs
+        .filter((doc) =>
+          this.hasPermissionsForDoc(
+            doc,
+            response.rows.find((responseDoc) => responseDoc.id === doc._id),
+            ability,
+          ),
+        )
+        .map((doc) =>
+          this.removeFieldsWithoutPermissions(
+            doc,
+            response.rows.find((responseDoc) => responseDoc.id === doc._id),
+            ability,
+          ),
         ),
-      ),
     };
   }
 
+  private deleteEmptyValues(
```

Comment:
For the review and later following the code I think it would help to sort the functions in a file according to their importance and call order. Having this function before `removeFieldsWithoutPermissions` doesn't really make sense.